optaplanner-core/src/main/java/org/optaplanner/core/impl/io/jaxb/GenericJaxbIO.java

```diff
@@ -0,0 +1,318 @@
+         */
+        @SuppressWarnings({ "java:S2755", "java:S4435" })
+        TransformerFactory transformerFactory = TransformerFactory.newInstance();
+        try {
+            Transformer transformer = transformationTemplate == null ? transformerFactory.newTransformer()
+                    : transformerFactory.newTransformer(transformationTemplate);
+            // See https://stackoverflow.com/questions/46708498/jaxb-marshaller-indentation.
+            transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+            transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", String.valueOf(indentation));
+            transformer.transform(source, new StreamResult(writer));
+        } catch (TransformerException transformerException) {
+            String errMessage = String.format(ERR_MSG_WRITE, rootClass.getName());
+            throw new OptaPlannerXmlSerializationException(errMessage, transformerException);
+        }
+    }
+
+    /**
+     * Overrides namespace of every XML element by the namespace defined in the {@link ElementNamespaceOverride}.
+     */
+    private static final class NamespaceOverridingXmlFilter extends XMLFilterImpl {
+
+        private final Deque<String> activeNamespace = new ArrayDeque<>();
+        private final Map<String, String> elementNamespaceOverridesMap = new HashMap<>();
+
+        public NamespaceOverridingXmlFilter(XMLReader xmlReader, ElementNamespaceOverride... elementNamespaceOverrides) {
+            super(xmlReader);
+            Objects.requireNonNull(elementNamespaceOverrides);
+            for (ElementNamespaceOverride namespaceOverride : elementNamespaceOverrides) {
+                elementNamespaceOverridesMap.put(namespaceOverride.getElementLocalName(),
+                        namespaceOverride.getNamespaceOverride());
+            }
+        }
+
+        @Override
+        public void endElement(String uri, String localName, String qName) throws SAXException {
+            String resultingUri = activeNamespace.isEmpty() ? uri : activeNamespace.peek();
+            if (elementNamespaceOverridesMap.containsKey(qName)) {
+                activeNamespace.pop();
+            }
+            super.endElement(resultingUri, localName, qName);
+        }
+
+        @Override
+        public void startElement(String uri, String localName, String qName, Attributes atts) throws SAXException {
+            String namespaceOverride = elementNamespaceOverridesMap.get(qName);
+            if (namespaceOverride != null) {
+                activeNamespace.push(namespaceOverride);
+            }
+
+            String resultingUri = activeNamespace.isEmpty() ? uri : activeNamespace.peek();
+            super.startElement(resultingUri, localName, qName, atts);
+        }
+
+    }
+
+    private static final class ValidationEventStringCollector extends ValidationEventCollector {
```

Comment:
Do we need a custom class for this?
We can probably just do 

`String s = validationEventCollector.getEvents().stream().map(Event::getMessage).collect(Collectors.joining("\n"))`